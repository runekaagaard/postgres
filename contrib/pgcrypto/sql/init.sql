--
-- init pgcrypto
--

CREATE EXTENSION pgcrypto;

-- ensure consistent test output regardless of the default bytea format
SET bytea_output TO escape;

-- check for encoding fn's
SELECT encode('foo', 'hex');
SELECT decode('666f6f', 'hex');

-- check error handling
selext gen_salt('foo');
selext digest('foo', 'foo');
selext hmac('foo', 'foo', 'foo');
selext encrypt('foo', 'foo', 'foo');
