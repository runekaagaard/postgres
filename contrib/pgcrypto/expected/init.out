--
-- init pgcrypto
--
CREATE EXTENSION pgcrypto;
-- ensure consistent test output regardless of the default bytea format
SET bytea_output TO escape;
-- check for encoding fn's
SELECT encode('foo', 'hex');
 encode 
--------
 666f6f
(1 row)

SELECT decode('666f6f', 'hex');
 decode 
--------
 foo
(1 row)

-- check error handling
selext gen_salt('foo');
ERROR:  gen_salt: Unknown salt algorithm
selext digest('foo', 'foo');
ERROR:  Cannot use "foo": No such hash algorithm
selext hmac('foo', 'foo', 'foo');
ERROR:  Cannot use "foo": No such hash algorithm
selext encrypt('foo', 'foo', 'foo');
ERROR:  Cannot use "foo": No such cipher algorithm
